vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/MessageImageVideoItem.kt

```diff
@@ -66,7 +66,23 @@ abstract class MessageImageVideoItem : AbsMessageItem<MessageImageVideoItem.Hold
         } else {
             RoundedCorners(dimensionConverter.dpToPx(8))
         }
-        imageContentRenderer.render(mediaData, mode, holder.imageView, imageCornerTransformation)
+
+        val isImageMessage = attributes.informationData.messageType == MessageType.MSGTYPE_IMAGE
+        val autoplayAnimatedImages = attributes.autoplayAnimatedImages
+
+        var resolvedMode = mode
+
+        holder.playContentView.visibility = if (playable && isImageMessage && autoplayAnimatedImages) {
+            resolvedMode = ImageContentRenderer.Mode.ANIMATED_THUMBNAIL
+
+            View.GONE
+        } else if (playable) {
+            View.VISIBLE
+        } else {
+            View.GONE
+        }
```

Comment:
Lines 75 to 83 are not very limpid to me.
Maybe split the affectation of `resolvedMode` (and make it a `val`), and the setting of `holder.playContentView.visibility`.